src/execution/fmt_impl.cpp

```diff
@@ -37,7 +37,8 @@ static auto HelperVecExprFmt(const std::vector<const AbstractExpression *> &expr
 
 auto AggregationPlanNode::PlanNodeToString() const -> std::string {
   auto aggregates = HelperVecExprFmt(aggregates_);
-  return fmt::format("Agg {{ types={}, aggregates=[{}] }}", agg_types_, aggregates);
+  return fmt::format("Agg {{ types={}, aggregates=[{}], having={}, group_by=[{}] }}", agg_types_, aggregates,
+                     having_ == nullptr ? "" : having_->ToString(), HelperVecExprFmt(group_bys_));
```

Comment:
IMO explicitly representing nullptr as "nullptr" is clearer, but up to you

Reply:
Same comment for the other places this happens.

Actually, the `blah == nullptr ? "" : blah->ToString()` is a common enough pattern that maybe it can be made a macro or util function somewhere. But that might be overdoing things

Reply:
Exactly! After I've made everything into `shared_ptr / unique_ptr`, I'll change this to:

```
fmt::format("having={}", having_)
```

We will have implementation of format `unique_ptr<Expression>`, which yields "nullptr" or "" automatically, without manually handling the case.